cmd/hubagent/workload/setup.go

```diff
@@ -119,6 +120,30 @@ func SetupControllers(ctx context.Context, mgr ctrl.Manager, config *rest.Config
 	if opts.EnableV1Beta1APIs {
 		klog.Info("Setting up clusterResourcePlacement v1beta1 controller")
 		clusterResourcePlacementControllerV1Beta1 = controller.NewController(crpControllerV1Beta1Name, controller.NamespaceKeyFunc, crpc.Reconcile, rateLimiter)
+
+		// Start up the controller.
+		//
+		// Note that, with v1alpha1 controllers, all custom ones are started up alongside the
+		// resource change detector, when its Start() method is invoked by the controller
+		// manager. This is no longer the case with v1beta1 controllers, specifically, the
+		// v1beta1 CRP controller, as it depends on its own watchers. As a result, here
+		// this controller is spun up separately.
+		//
+		// The controller must run in a separate goroutine as Run() is a blocking call.
```

Comment:
This reconciler is added to the mgr at line 150 (old line). Is that not enough?

Reply:
Hi Ryan! The one at line 150 is the watcher for the v1beta1 CRP controller (it pushes keys to the workqueue); the v1beta1 CRP controller is not started at this moment (its Run() is not called).